### DISCUSSION TITLE
Can I use tools from this repo to create an online blazor-driven experience?

### DISCUSSION THREAD

**Discussion by douglasg14b** [dotnet/interactive#2931] [Q&A]
I read that this repo is a replacement of `try.net`, which had online editor experiences, but was not really built to enable other devs to use it's tooling.

Example of an online editor: https://learn.microsoft.com/en-us/training/modules/csharp-write-first/2-exercise-hello-world

I want to create an online experience with notebook blocks, and interactive editors such as this to showcase C# features as part of a presentation, so colleagues can follow along and have a guided playground to use without any more friction than opening a webpage.

Do you have a recommended direction to achieve this?

**Comment by HowardvanRooijen** [comment]
It's not a replacement - .NET Interactive is a core "engine" that powers Try .NET (https://github.com/dotnet/try) as well as Polyglot Notebooks. Try .NET powers the Microsoft Learn examples, and NodaTime (see https://nodatime.org/3.1.x/userguide/recipes). 

You can set up your own instance of Try .NET but it requires running an ASP.NET backend and hosting the WASM editor. AFAIK there isn't an OOB sample that provides this for you. You scenario is something I've been interested in for the last couple of years.

Have you thought about using Polyglot Notebooks, which you can run via CodeSpaces?

**Reply by douglasg14b** [reply]
CodeSpaces was a thought as well, but it's higher friction and slower to wind up than a normal site/app.

I can setup Try.Net and check it out directly, I haven't as of yet since slogging through someone else's codebase to extract features I want to use is a rough task. Plus I can't find the source for the bit that really interests me, the actual .Net docs style editor 😞 

<hr>

Do you by chance know where the source for just the editor part of the link in the OP is? I did some searching for text copies, elements, outputs, and css classes but have not really found it yet. It doesn't appear to be in the Try .NET repo 🤔 

It seems to be in an i-frame that points to `https://trydotnet.microsoft.com/v2/editor?waitForConfiguration=true&editorId=0&workspaceType=console`, but I can't seem to find the actual source. Github global search shows nothing for the CSS classes there like `dot-net-style-app-editor` and `mscomdocs-style-app-editor`.

**Reply by jonsequitur** [reply]
The code for the Try. NET service was never released as open source. As Howard said though, the core source code is in this repository. 

We're in the process of making some improvements here that should make it much easier for people to host Try .NET.  We'll post updates as we make progress.
